pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -1729,6 +1729,7 @@ static void updateColumnNames(String rawTableName, PinotQuery pinotQuery, boolea
           hasStar = true;
         }
       }
+      pinotQuery.setSelectStarQuery(hasStar);
```

Comment:
Can you test how this works in Multistage query execution? The problem should remain the same during leaf-stage processing. We should fix it there as well. Or at least leave a TODO.